File: bd/java/inv-dao/src/cu/cenpis/gps/inv/data/entity/ActivoFijo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cu.cenpis.gps.inv.data.entity;

import java.io.Serializable;
import java.util.Date;
import javax.persistence.Basic;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.Lob;
import javax.persistence.ManyToOne;
import javax.persistence.NamedQueries;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import javax.xml.bind.annotation.XmlRootElement;

/**
 *
 * @author farias-i5
 */
@Entity
@Table(name = "activo_fijo", schema = "")
@XmlRootElement
@NamedQueries({
    @NamedQuery(name = "ActivoFijo.findAll", query = "SELECT a FROM ActivoFijo a"),
    @NamedQuery(name = "ActivoFijo.findByIdActivoFijo", query = "SELECT a FROM ActivoFijo a WHERE a.idActivoFijo = :idActivoFijo"),
    @NamedQuery(name = "ActivoFijo.findByRotulo", query = "SELECT a FROM ActivoFijo a WHERE a.rotulo = :rotulo"),
    @NamedQuery(name = "ActivoFijo.findByValorCuc", query = "SELECT a FROM ActivoFijo a WHERE a.valorCuc = :valorCuc"),
    @NamedQuery(name = "ActivoFijo.findByValorMn", query = "SELECT a FROM ActivoFijo a WHERE a.valorMn = :valorMn"),
    @NamedQuery(name = "ActivoFijo.findByTasa", query = "SELECT a FROM ActivoFijo a WHERE a.tasa = :tasa"),
    @NamedQuery(name = "ActivoFijo.findByDepAcuCuc", query = "SELECT a FROM ActivoFijo a WHERE a.depAcuCuc = :depAcuCuc"),
    @NamedQuery(name = "ActivoFijo.findByDepAcuMn", query = "SELECT a FROM ActivoFijo a WHERE a.depAcuMn = :depAcuMn"),
    @NamedQuery(name = "ActivoFijo.findByValorActualCuc", query = "SELECT a FROM ActivoFijo a WHERE a.valorActualCuc = :valorActualCuc"),
    @NamedQuery(name = "ActivoFijo.findByValorActualMn", query = "SELECT a FROM ActivoFijo a WHERE a.valorActualMn = :valorActualMn"),
    @NamedQuery(name = "ActivoFijo.findByResponsableText", query = "SELECT a FROM ActivoFijo a WHERE a.responsableText = :responsableText"),
    @NamedQuery(name = "ActivoFijo.findByEstadoText", query = "SELECT a FROM ActivoFijo a WHERE a.estadoText = :estadoText"),
    @NamedQuery(name = "ActivoFijo.findByFechaAlta", query = "SELECT a FROM ActivoFijo a WHERE a.fechaAlta = :fechaAlta"),
    @NamedQuery(name = "ActivoFijo.findByFechaEstadoActual", query = "SELECT a FROM ActivoFijo a WHERE a.fechaEstadoActual = :fechaEstadoActual"),
    @NamedQuery(name = "ActivoFijo.findRevision", query = "SELECT a FROM ActivoFijo a WHERE a.rotulo = :mRotulo AND a.revision.idRevision = :idRevision"),
    @NamedQuery(name = "ActivoFijo.findByRevisionActiva", query = "SELECT a FROM ActivoFijo a WHERE a.revision.activo = 1"),
    @NamedQuery(name = "ActivoFijo.findByPrestado", query = "SELECT a FROM ActivoFijo a WHERE a.prestado = :prestado"),
    @NamedQuery(name = "ActivoFijo.findByProcesoBaja", query = "SELECT a FROM ActivoFijo a WHERE a.procesoBaja = :procesoBaja")
})
public class ActivoFijo implements Serializable {

    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Basic(optional = false)
    @Column(name = "id_activo_fijo")
    private Long idActivoFijo;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 18)
    private String rotulo;
    @Basic(optional = false)
    @NotNull
    @Lob
    @Size(min = 1, max = 65535)
    private String descripcion;
    @Basic(optional = false)
    @NotNull
    @Column(name = "valor_cuc")
    private Float valorCuc;
    @Basic(optional = false)
    @NotNull
    @Column(name = "valor_mn")
    private Float valorMn;
    @Basic(optional = false)
    @NotNull
    private Float tasa;
    @Basic(optional = false)
    @NotNull
    @Column(name = "dep_acu_cuc")
    private Float depAcuCuc;
    @Basic(optional = false)
    @NotNull
    @Column(name = "dep_acu_mn")
    private Float depAcuMn;
    @Basic(optional = false)
    @NotNull
    @Column(name = "valor_actual_cuc")
    private Float valorActualCuc;
    @Basic(optional = false)
    @NotNull
    @Column(name = "valor_actual_mn")
    private Float valorActualMn;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 100)
    @Column(name = "responsable_text")
    private String responsableText;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 100)
    @Column(name = "estado_text")
    private String estadoText;
    @Basic(optional = false)
    @NotNull
    @Column(name = "fecha_alta")
    @Temporal(TemporalType.DATE)
    private Date fechaAlta;
    @Basic(optional = false)
    @NotNull
    @Column(name = "fecha_estado_actual")
    @Temporal(TemporalType.DATE)
    private Date fechaEstadoActual;
    @Basic(optional = false)
    @NotNull
    private boolean prestado;
    @Basic(optional = false)
    @NotNull
    @Column(name = "proceso_baja")
    private boolean procesoBaja;
    @JoinColumn(name = "estado", referencedColumnName = "id_estado")
    @ManyToOne(optional = false)
    private Estado estado;
    @JoinColumn(name = "local", referencedColumnName = "id_local")
    @ManyToOne(optional = false)
    private Local local;
    @JoinColumn(name = "responsable", referencedColumnName = "id_responsable")
    @ManyToOne(optional = false)
    private Responsable responsable;
    @JoinColumn(name = "revision", referencedColumnName = "id_revision")
    @ManyToOne(optional = false)
    private Revision revision;
    @JoinColumn(name = "tipo_activo", referencedColumnName = "id_tipo_activo")
    @ManyToOne
    private TipoActivo tipoActivo;
    
    public ActivoFijo() {
    }

    public ActivoFijo(String rotulo, String descripcion, Float valorCuc, Float valorMn,
            Float tasa, Float depAcuCuc, Float depAcuMn, Float valorActualCuc,
            Float valorActualMn, String responsableText, String estadoText,
            Date fechaAlta, Date fechaEstadoActual, Estado estado, Local local,
            Responsable responsable, Revision revision, TipoActivo tipoActivo) {
        this.rotulo = rotulo;
        this.descripcion = descripcion;
        this.valorCuc = valorCuc;
        this.valorMn = valorMn;
        this.tasa = tasa;
        this.depAcuCuc = depAcuCuc;
        this.depAcuMn = depAcuMn;
        this.valorActualCuc = valorActualCuc;
        this.valorActualMn = valorActualMn;
        this.responsableText = responsableText;
        this.estadoText = estadoText;
        this.fechaAlta = fechaAlta;
        this.fechaEstadoActual = fechaEstadoActual;
        this.estado = estado;
        this.local = local;
        this.responsable = responsable;
        this.revision = revision;
        this.tipoActivo = tipoActivo;
        this.prestado = false;
        this.procesoBaja = false;
    }

    public Long getIdActivoFijo() {
        return idActivoFijo;
    }

    public void setIdActivoFijo(Long idActivoFijo) {
        this.idActivoFijo = idActivoFijo;
    }

    public String getRotulo() {
        return rotulo;
    }

    public void setRotulo(String rotulo) {
        this.rotulo = rotulo;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public Float getValorCuc() {
        return valorCuc;
    }

    public void setValorCuc(Float valorCuc) {
        this.valorCuc = valorCuc;
    }

    public Float getValorMn() {
        return valorMn;
    }

    public void setValorMn(Float valorMn) {
        this.valorMn = valorMn;
    }

    public Float getTasa() {
        return tasa;
    }

    public void setTasa(Float tasa) {
        this.tasa = tasa;
    }

    public Float getDepAcuCuc() {
        return depAcuCuc;
    }

    public void setDepAcuCuc(Float depAcuCuc) {
        this.depAcuCuc = depAcuCuc;
    }

    public Float getDepAcuMn() {
        return depAcuMn;
    }

    public void setDepAcuMn(Float depAcuMn) {
        this.depAcuMn = depAcuMn;
    }

    public Float getValorActualCuc() {
        return valorActualCuc;
    }

    public void setValorActualCuc(Float valorActualCuc) {
        this.valorActualCuc = valorActualCuc;
    }

    public Float getValorActualMn() {
        return valorActualMn;
    }

    public void setValorActualMn(Float valorActualMn) {
        this.valorActualMn = valorActualMn;
    }

    public String getResponsableText() {
        return responsableText;
    }

    public void setResponsableText(String responsableText) {
        this.responsableText = responsableText;
    }

    public String getEstadoText() {
        return estadoText;
    }

    public void setEstadoText(String estadoText) {
        this.estadoText = estadoText;
    }

    public Date getFechaAlta() {
        return fechaAlta;
    }

    public void setFechaAlta(Date fechaAlta) {
        this.fechaAlta = fechaAlta;
    }

    public Date getFechaEstadoActual() {
        return fechaEstadoActual;
    }

    public void setFechaEstadoActual(Date fechaEstadoActual) {
        this.fechaEstadoActual = fechaEstadoActual;
    }

    public Estado getEstado() {
        return estado;
    }

    public void setEstado(Estado estado) {
        this.estado = estado;
    }

    public Local getLocal() {
        return local;
    }

    public void setLocal(Local local) {
        this.local = local;
    }

    public Responsable getResponsable() {
        return responsable;
    }

    public void setResponsable(Responsable responsable) {
        this.responsable = responsable;
    }

    public Revision getRevision() {
        return revision;
    }

    public void setRevision(Revision revision) {
        this.revision = revision;
    }

    public TipoActivo getTipoActivo() {
        return tipoActivo;
    }

    public void setTipoActivo(TipoActivo tipoActivo) {
        this.tipoActivo = tipoActivo;
    }

    public Boolean getPrestado() {
        return prestado;
    }

    public void setPrestado(Boolean prestado) {
        this.prestado = prestado;
    }

    public Boolean getProcesoBaja() {
        return procesoBaja;
    }

    public void setProcesoBaja(Boolean procesoBaja) {
        this.procesoBaja = procesoBaja;
    }

    @Override
    public int hashCode() {
        int hash = 0;
        hash += (idActivoFijo != null ? idActivoFijo.hashCode() : 0);
        return hash;
    }

    @Override
    public boolean equals(Object object) {
        // TODO: Warning - this method won't work in the case the id fields are not set
        if (!(object instanceof ActivoFijo)) {
            return false;
        }
        ActivoFijo other = (ActivoFijo) object;
        if ((this.idActivoFijo == null && other.idActivoFijo != null) || (this.idActivoFijo != null && !this.idActivoFijo.equals(other.idActivoFijo))) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "cu.ActivoFijo[ idActivoFijo=" + idActivoFijo + " ]";
    }

}

File: src/INV/CommonBundle/Entity/ActivoFijo.php
<?php

namespace INV\CommonBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * ActivoFijo
 *
 * @ORM\Table(name="activo_fijo")
 * @ORM\Entity(repositoryClass="INV\CommonBundle\Repository\ActivoFijoRepository")
 */
class ActivoFijo {
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="bigint", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="rotulo", type="string", length=50, nullable=false)
     */
    private $rotulo;

    /**
     * @var string
     *
     * @ORM\Column(name="descripcion", type="text", length=65535, nullable=false)
     */
    private $descripcion;

    /**
     * @var float
     *
     * @ORM\Column(name="valor_cuc", type="float", precision=9, scale=3, nullable=false)
     */
    private $valorCuc;

    /**
     * @var float
     *
     * @ORM\Column(name="valor_mn", type="float", precision=9, scale=3, nullable=false)
     */
    private $valorMn;

    /**
     * @var float
     *
     * @ORM\Column(name="tasa", type="float", precision=9, scale=3, nullable=false)
     */
    private $tasa;

    /**
     * @var float
     *
     * @ORM\Column(name="dep_acu_cuc", type="float", precision=9, scale=3, nullable=false)
     */
    private $depAcuCuc;

    /**
     * @var float
     *
     * @ORM\Column(name="dep_acu_mn", type="float", precision=9, scale=3, nullable=false)
     */
    private $depAcuMn;

    /**
     * @var float
     *
     * @ORM\Column(name="valor_actual_cuc", type="float", precision=9, scale=3, nullable=false)
     */
    private $valorActualCuc;

    /**
     * @var float
     *
     * @ORM\Column(name="valor_actual_mn", type="float", precision=9, scale=3, nullable=false)
     */
    private $valorActualMn;

    /**
     * @var string
     *
     * @ORM\Column(name="responsable_text", type="string", length=100, nullable=false)
     */
    private $responsableText;

    /**
     * @var string
     *
     * @ORM\Column(name="estado_text", type="string", length=100, nullable=false)
     */
    private $estadoText;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="fecha_alta", type="date", nullable=false)
     */
    private $fechaAlta;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="fecha_estado_actual", type="date", nullable=false)
     */
    private $fechaEstadoActual;

    /**
     * @var Estado
     *
     * @ORM\ManyToOne(targetEntity="Estado")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="estado", referencedColumnName="id", nullable=false)
     * })
     */
    private $estado;

    /**
     * @var Local
     *
     * @ORM\ManyToOne(targetEntity="Local")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="local", referencedColumnName="id", nullable=false)
     * })
     */
    private $local;

    /**
     * @var Responsable
     *
     * @ORM\ManyToOne(targetEntity="Responsable")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="responsable", referencedColumnName="id", nullable=false)
     * })
     */
    private $responsable;

    /**
     * @var Revision
     *
     * @ORM\ManyToOne(targetEntity="Revision")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="revision", referencedColumnName="id", nullable=false)
     * })
     */
    private $revision;

    /**
     * @var TipoActivo
     *
     * @ORM\ManyToOne(targetEntity="TipoActivo")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="tipo_activo", referencedColumnName="id", nullable=false)
     * })
     */
    private $tipoActivo;

    /**
     * @var string
     *
     * @ORM\Column(name="serie", type="string", length=255, nullable=true)
     */
    private $serie;

    /**
     * @var string
     *
     * @ORM\Column(name="marca", type="string", length=255, nullable=true)
     */
    private $marca;

    /**
     * @var string
     *
     * @ORM\Column(name="modelo", type="string", length=255, nullable=true)
     */
    private $modelo;

    /**
     * @var string
     *
     * @ORM\Column(name="pais_procedencia", type="string", length=255, nullable=true)
     */
    private $paisProcedencia;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="fecha_produccion", type="date", nullable=true)
     */
    private $fechaProduccion;

    /**
     * @var \Doctrine\Common\Collections\Collection
     *
     * @ORM\ManyToMany(targetEntity="INV\CommonBundle\Entity\Informe", inversedBy="activos")
     * @ORM\JoinTable(name="informe_activo_fijo",
     *   joinColumns={
     *     @ORM\JoinColumn(name="activo_fijo", referencedColumnName="id")
     *   },
     *   inverseJoinColumns={
     *     @ORM\JoinColumn(name="informe", referencedColumnName="id")
     *   }
     * )
     */
    private $informes;

    function __construct() {
        $this->informes = new \Doctrine\Common\Collections\ArrayCollection();
    }


    /**
     * Get id.
     *
     * @return int
     */
    public function getId() {
        return $this->id;
    }

    /**
     * Set rotulo.
     *
     * @param string $rotulo
     *
     * @return ActivoFijo
     */
    public function setRotulo($rotulo) {
        $this->rotulo = $rotulo;

        return $this;
    }

    /**
     * Get rotulo.
     *
     * @return string
     */
    public function getRotulo() {
        return $this->rotulo;
    }

    /**
     * Set descripcion.
     *
     * @param string $descripcion
     *
     * @return ActivoFijo
     */
    public function setDescripcion($descripcion) {
        $this->descripcion = $descripcion;

        return $this;
    }

    /**
     * Get descripcion.
     *
     * @return string
     */
    public function getDescripcion() {
        return $this->descripcion;
    }

    /**
     * Set valorCuc.
     *
     * @param float $valorCuc
     *
     * @return ActivoFijo
     */
    public function setValorCuc($valorCuc) {
        $this->valorCuc = $valorCuc;

        return $this;
    }

    /**
     * Get valorCuc.
     *
     * @return float
     */
    public function getValorCuc() {
        return $this->valorCuc;
    }

    /**
     * Set valorMn.
     *
     * @param float $valorMn
     *
     * @return ActivoFijo
     */
    public function setValorMn($valorMn) {
        $this->valorMn = $valorMn;

        return $this;
    }

    /**
     * Get valorMn.
     *
     * @return float
     */
    public function getValorMn() {
        return $this->valorMn;
    }

    /**
     * Set tasa.
     *
     * @param float $tasa
     *
     * @return ActivoFijo
     */
    public function setTasa($tasa) {
        $this->tasa = $tasa;

        return $this;
    }

    /**
     * Get tasa.
     *
     * @return float
     */
    public function getTasa() {
        return $this->tasa;
    }

    /**
     * Set depAcuCuc.
     *
     * @param float $depAcuCuc
     *
     * @return ActivoFijo
     */
    public function setDepAcuCuc($depAcuCuc) {
        $this->depAcuCuc = $depAcuCuc;

        return $this;
    }

    /**
     * Get depAcuCuc.
     *
     * @return float
     */
    public function getDepAcuCuc() {
        return $this->depAcuCuc;
    }

    /**
     * Set depAcuMn.
     *
     * @param float $depAcuMn
     *
     * @return ActivoFijo
     */
    public function setDepAcuMn($depAcuMn) {
        $this->depAcuMn = $depAcuMn;

        return $this;
    }

    /**
     * Get depAcuMn.
     *
     * @return float
     */
    public function getDepAcuMn() {
        return $this->depAcuMn;
    }

    /**
     * Set valorActualCuc.
     *
     * @param float $valorActualCuc
     *
     * @return ActivoFijo
     */
    public function setValorActualCuc($valorActualCuc) {
        $this->valorActualCuc = $valorActualCuc;

        return $this;
    }

    /**
     * Get valorActualCuc.
     *
     * @return float
     */
    public function getValorActualCuc() {
        return $this->valorActualCuc;
    }

    /**
     * Set valorActualMn.
     *
     * @param float $valorActualMn
     *
     * @return ActivoFijo
     */
    public function setValorActualMn($valorActualMn) {
        $this->valorActualMn = $valorActualMn;

        return $this;
    }

    /**
     * Get valorActualMn.
     *
     * @return float
     */
    public function getValorActualMn() {
        return $this->valorActualMn;
    }

    /**
     * Set responsableText.
     *
     * @param string $responsableText
     *
     * @return ActivoFijo
     */
    public function setResponsableText($responsableText) {
        $this->responsableText = $responsableText;

        return $this;
    }

    /**
     * Get responsableText.
     *
     * @return string
     */
    public function getResponsableText() {
        return $this->responsableText;
    }

    /**
     * Set estadoText.
     *
     * @param string $estadoText
     *
     * @return ActivoFijo
     */
    public function setEstadoText($estadoText) {
        $this->estadoText = $estadoText;

        return $this;
    }

    /**
     * Get estadoText.
     *
     * @return string
     */
    public function getEstadoText() {
        return $this->estadoText;
    }

    /**
     * Set fechaAlta.
     *
     * @param \DateTime $fechaAlta
     *
     * @return ActivoFijo
     */
    public function setFechaAlta($fechaAlta) {
        $this->fechaAlta = $fechaAlta;

        return $this;
    }

    /**
     * Get fechaAlta.
     *
     * @return \DateTime
     */
    public function getFechaAlta() {
        return $this->fechaAlta;
    }

    /**
     * Set fechaEstadoActual.
     *
     * @param \DateTime $fechaEstadoActual
     *
     * @return ActivoFijo
     */
    public function setFechaEstadoActual($fechaEstadoActual) {
        $this->fechaEstadoActual = $fechaEstadoActual;

        return $this;
    }

    /**
     * Get fechaEstadoActual.
     *
     * @return \DateTime
     */
    public function getFechaEstadoActual() {
        return $this->fechaEstadoActual;
    }

    /**
     * Set estado.
     *
     * @param \INV\CommonBundle\Entity\Estado|null $estado
     *
     * @return ActivoFijo
     */
    public function setEstado(\INV\CommonBundle\Entity\Estado $estado = null) {
        $this->estado = $estado;

        return $this;
    }

    /**
     * Get estado.
     *
     * @return \INV\CommonBundle\Entity\Estado|null
     */
    public function getEstado() {
        return $this->estado;
    }

    /**
     * Set local.
     *
     * @param \INV\CommonBundle\Entity\Local|null $local
     *
     * @return ActivoFijo
     */
    public function setLocal(\INV\CommonBundle\Entity\Local $local = null) {
        $this->local = $local;

        return $this;
    }

    /**
     * Get local.
     *
     * @return \INV\CommonBundle\Entity\Local|null
     */
    public function getLocal() {
        return $this->local;
    }

    /**
     * Set responsable.
     *
     * @param \INV\CommonBundle\Entity\Responsable|null $responsable
     *
     * @return ActivoFijo
     */
    public function setResponsable(\INV\CommonBundle\Entity\Responsable $responsable = null) {
        $this->responsable = $responsable;

        return $this;
    }

    /**
     * Get responsable.
     *
     * @return \INV\CommonBundle\Entity\Responsable|null
     */
    public function getResponsable() {
        return $this->responsable;
    }

    /**
     * Set revision.
     *
     * @param \INV\CommonBundle\Entity\Revision|null $revision
     *
     * @return ActivoFijo
     */
    public function setRevision(\INV\CommonBundle\Entity\Revision $revision = null) {
        $this->revision = $revision;

        return $this;
    }

    /**
     * Get revision.
     *
     * @return \INV\CommonBundle\Entity\Revision|null
     */
    public function getRevision() {
        return $this->revision;
    }

    /**
     * Set tipoActivo.
     *
     * @param \INV\CommonBundle\Entity\TipoActivo|null $tipoActivo
     *
     * @return ActivoFijo
     */
    public function setTipoActivo(\INV\CommonBundle\Entity\TipoActivo $tipoActivo = null) {
        $this->tipoActivo = $tipoActivo;

        return $this;
    }

    /**
     * Get tipoActivo.
     *
     * @return \INV\CommonBundle\Entity\TipoActivo|null
     */
    public function getTipoActivo() {
        return $this->tipoActivo;
    }

    /**
     * Add informe
     *
     * @param Informe $informe
     *
     * @return ActivoFijo
     */
    public function addInforme(Informe $informe) {
        $this->informes[] = $informe;

        return $this;
    }

    /**
     * Remove informe
     *
     * @param Informe $informe
     */
    public function removeInforme(Informe $informe) {
        $this->informes->removeElement($informe);
    }

    /**
     * Get informes
     *
     * @return \Doctrine\Common\Collections\Collection
     */
    public function getInformes() {
        return $this->informes;
    }

    /**
     * @return string
     */
    public function getSerie() {
        return $this->serie;
    }

    /**
     * @param string $serie
     */
    public function setSerie($serie) {
        $this->serie = $serie;
    }

    /**
     * @return string
     */
    public function getModelo() {
        return $this->modelo;
    }

    /**
     * @param string $modelo
     */
    public function setModelo($modelo) {
        $this->modelo = $modelo;
    }

    /**
     * @return string
     */
    public function getPaisProcedencia() {
        return $this->paisProcedencia;
    }

    /**
     * @param string $paisProcedencia
     */
    public function setPaisProcedencia($paisProcedencia) {
        $this->paisProcedencia = $paisProcedencia;
    }

    /**
     * @return \DateTime
     */
    public function getFechaProduccion() {
        return $this->fechaProduccion;
    }

    /**
     * @param \DateTime $fechaProduccion
     */
    public function setFechaProduccion($fechaProduccion) {
        $this->fechaProduccion = $fechaProduccion;
    }

    /**
     * @return string
     */
    public function getMarca() {
        return $this->marca;
    }

    /**
     * @param string $marca
     */
    public function setMarca($marca) {
        $this->marca = $marca;
    }

    public function toArray() {
        $array = array();
        $array[] = $this->getDescripcion();
        $array[] = $this->getRotulo();
        $array[] = $this->getSerie();
        $array[] = $this->getModelo();
        $array[] = $this->getFechaAlta()->format('Y-m-d');
        $array[] = $this->getPaisProcedencia();
        $array[] = $this->getValorActualCuc();
        $array[] = $this->getValorActualMn();
        $array[] = "Docencia-Investigación";
        $array[] = $this->getFechaEstadoActual()->format('Y-m-d');
        $array[] = "2094034";
        return $array;
    }
}
